src/inclusion_proof.rs

```diff
@@ -1,53 +1,57 @@
+    HeaderWithProof,
+};
+
 use primitive_types::H256;
+use sf_protos::ethereum::r#type::v2::Block;
 use trin_validation::accumulator::MasterAccumulator;
 
-// function: generate_inclusion_proof
-// inputs: flat_file_directory, master_accumulator_file, start_block, end_block
-// outputs: inclusion_proof
+/// generates an inclusion proof over headers, given blocks between `start_block` and `end_block`
+///
+/// # Arguments
+///
+/// * `ext_headers`-  A mutable [`Vec<ExtHeaderRecord>`]. The Vector can be any size, however, it must be in chunks of 8192 blocks to work properly
+/// to function without error
+/// * `start_block` -  The starting point of blocks that are to be included in the proofs. This interval is inclusive.
+/// * `end_epoch` -  The ending point of blocks that are to be included in the proofs. This interval is inclusive.
 pub fn generate_inclusion_proof(
-    directory: &String,
-    start_block: usize,
-    end_block: usize,
+    mut ext_headers: Vec<ExtHeaderRecord>,
+    start_block: u64,
+    end_block: u64,
 ) -> Result<Vec<[H256; 15]>, EraValidateError> {
     // Compute the epoch accumulator for the blocks
     // The epochs start on a multiple of 8192 blocks, so we need to round down to the nearest 8192
     let epoch_start = start_block / 8192;
 
     // The epochs end on a multiple of 8192 blocks, so we need to round up to the nearest 8192
-    let epoch_end = ((end_block as f32) / 8192.0).ceil() as usize;
+    let epoch_end = ((end_block as f32) / 8192.0).ceil() as u64;
 
     // We need to load blocks from an entire epoch to be able to generate inclusion proofs
     // First compute epoch accumulators and the Merkle tree for all the epochs of interest
+    // let mut epoch_accumulators: Vec<_> = Vec::new();
     let mut epoch_accumulators = Vec::new();
-    let mut inclusion_proof_vec = Vec::new();
-    let mut headers = Vec::new();
-    for epoch in epoch_start..epoch_end {
-        let start_block = epoch * 8192;
-        let end_block = (epoch + 1) * 8192;
+    let mut inclusion_proof_vec: Vec<[H256; 15]> = Vec::new();
+    let mut headers: Vec<Header> = Vec::new();
 
-        let blocks = extract_100_blocks(directory, start_block, end_block)?;
-
-        let mut blocks_headers = Vec::new();
-        for block in blocks.clone() {
-            let header = header_from_block(block)?;
-            blocks_headers.push(header.clone());
-        }
-        let header_records = decode_header_records(blocks)?;
-        headers.extend(blocks_headers);
+    for _ in epoch_start..epoch_end {
+        let epoch_headers: Vec<ExtHeaderRecord> = ext_headers.drain(0..MAX_EPOCH_SIZE).collect();
+        let header_records: Vec<HeaderRecord> = epoch_headers.iter().map(Into::into).collect();
+        let tmp_headers: Vec<Header> = epoch_headers.into_iter().map(Into::into).collect();
+        headers.extend(tmp_headers);
         epoch_accumulators.push(compute_epoch_accumulator(&header_records)?);
```

Comment:
I think this is fine for now, but in the future, it might be better to pass `epoch_accumulators` as an input to this function than to construct them here.